configure.ac

```diff
@@ -2075,19 +2075,75 @@ dnl **********************************************************************
 dnl Check for libpsl
 dnl **********************************************************************
 
-AC_ARG_WITH(libpsl,
-           AS_HELP_STRING([--without-libpsl],
-           [disable support for libpsl]),
-           with_libpsl=$withval,
-           with_libpsl=yes)
-curl_psl_msg="no      (libpsl disabled)"
-if test $with_libpsl != "no"; then
-  AC_SEARCH_LIBS(psl_builtin, psl,
-    [curl_psl_msg="enabled";
-     AC_DEFINE([USE_LIBPSL], [1], [PSL support enabled])
-     ],
-    [AC_MSG_ERROR([libpsl was not found]) ]
+dnl Default to compiler & linker defaults for LIBPSL files & libraries.
+OPT_LIBPSL=off
+AC_ARG_WITH(libpsl,dnl
+AS_HELP_STRING([--with-libpsl=PATH],[Where to look for libpsl, PATH points to the LIBPSL installation; when possible, set the PKG_CONFIG_PATH environment variable instead of using this option])
+AS_HELP_STRING([--without-libpsl], [disable LIBPSL]),
+  OPT_LIBPSL=$withval)
+
+if test X"$OPT_LIBPSL" != Xno; then
+  dnl backup the pre-libpsl variables
+  CLEANLDFLAGS="$LDFLAGS"
+  CLEANCPPFLAGS="$CPPFLAGS"
+  CLEANLIBS="$LIBS"
+
+  case "$OPT_LIBPSL" in
+  yes)
+    dnl --with-libpsl (without path) used
+    CURL_CHECK_PKGCONFIG(libpsl)
+
+    if test "$PKGCONFIG" != "no" ; then
+      LIB_PSL=`$PKGCONFIG --libs-only-l libpsl`
+      LD_PSL=`$PKGCONFIG --libs-only-L libpsl`
+      CPP_PSL=`$PKGCONFIG --cflags-only-I libpsl`
+    else
+      dnl To avoid link errors, we do not allow --libpsl without
+      dnl a pkgconfig file
+      AC_MSG_ERROR([--libpsl was specified but could not find libpsl pkgconfig file.])
```

Comment:
I assume you mean --with-libpsl and why would we not want to allow it without pkg-config?

Reply:
I'll rename `--libpsl` to `--with-libpsl`.
If `--with-libpsl` is provided, but pkg-config file is not found, an error is raised as already done by libcurl when handling librtmp, nghttp2, ngtcp2, nghttp3 and quiche. However, it seems that libcurl is not very consistent as other options don't raise such kind of failure.

Reply:
It is not consistent as to whether that is an error or not, correct. Like in this case, curl has built fine with libpsl for decades without pkg-config, so maybe it could be worth keeping that ability just to make existing setups keep working?

Reply:
PR was updated